src/stt/analyze_data.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+
+"""
+Description:
+    Creating a PDF report about the features of the dataset
+
+Usage:
+    $ python src/data/analyze_data.py
+"""
+import os
+import sys
+
+import librosa
+import numpy as np
+import pandas as pd
+from constants import (
+    RAW_RECORDINGS_DIR,
+    RECORDINGS_FILE,
+    SRC_DIR,
+    STT_REPORT,
+)
+from pandas_profiling import ProfileReport
+
+sys.path.insert(0, SRC_DIR)
+from decorators import log_function_name  # noqa: E402
+
+# ! Todo
+# # Load files
+# audio_segment = AudioSegment.from_file(
+#     "Downloads/Warm-Memories-Emotional-Inspiring-Piano.wav"
+# )
+# # Print attributes
+# print(f"Channels: {audio_segment.channels}")
+# print(f"Sample width: {audio_segment.sample_width}")
+# print(f"Frame rate (sample rate): {audio_segment.frame_rate}")
+# print(f"Frame width: {audio_segment.frame_width}")
+# print(f"Length (ms): {len(audio_segment)}")
+# print(f"Frame count: {audio_segment.frame_count()}")
+# print(f"Intensity: {audio_segment.dBFS}")
+# print(f"Get duration: {audio_segment.duration_seconds}")
+# # librsa.feature.mfcc
+
+
+def get_librosa_features(row: pd.Series) -> tuple[int, float]:
+    """
+    Get the duration of a wav file
+
+    Parameters
+    ----------
+    row : pd.Series
+        Row of the dataframe
+
+    Returns
+    -------
+    tuple[int, float]
+        Sample rate and duration of the wav file
+    """
+    file_path = row.file_location
+    sr = librosa.get_samplerate(file_path)
+    duration = librosa.get_duration(filename=file_path)
+    return sr, np.round(duration, 3)
+
+
+@log_function_name
```

Comment:
wofür steht immer @log_function_name über den Funktionen?

Reply:
Das erlaubt es mir anzuzeigen, welche Funktion gerade ausgeführt wird. In src/decorators.py habe ich die Funktion erstellt die das durchläuft